### ISSUE TITLE
Allow frame accurate events

### ISSUE THREAD

**Issue by trobanga** [tesselode/kira#75]
I noticed that `Clock::shared` is only updated in `on_process_start`, which causes a delay up to a few milliseconds e.g. in `ClockInfoProvider::when_to_start`.

I moved the update of  `Clock::shared` into `Clock::update` and I didn't notice a performance impact. On my machine, running `cargo bench` has a performance decrease of 3% for `simple` and (interestingly) almost no impact for `with on_start_processing callback`.

Do you think it's possible to move it there or maybe enable it with a feature? If you agree I'd be happy to make a PR.

**Comment by tesselode** [comment]
Hi! I noticed that too recently. That's a regression from 0.7! 🥶 I fixed it in 338112e, so it'll be in Kira v0.9 (I may even release a v0.8.7 with the fix if enough people care).

The way I fixed it is a little different - I just had the `ClockInfoProvider` grab the info directly from the clock instead of `ClockShared`. `ClockShared` is just for monitoring information from the gameplay thread, there's no reason the audio thread has to rely on it.
